### PR TITLE
Update to 0.58.1

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- Update to 0.58.1

https://github.com/numba/numba/compare/0.58.0...0.58.1

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -79,10 +77,9 @@ test:
     - pip
     - wheel
     - tbb >=2021.6,<2022
-    - llvm-openmp              # [osx]
```
Comment (markan) [inline recipe/meta.yaml:82]: Curiosity question: What changed allowing this to be removed? When I read setup.py there isn't much recent change around the OpenMP requirements for darwin.

Comment (cbouss) [reply recipe/meta.yaml:82]: It's redundant as llvm-openmp is already a run dependency of numba (through the implicit strong run export triggered by its inclusion in the build deps).

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -38,18 +38,16 @@ requirements:
...
     - _openmp_mutex 5.1        # [linux]
     - importlib_metadata  # [py<39]
   run:
     - python
     - importlib_metadata  # [py<39]
-    - {{ pin_compatible('llvmlite', max_pin='x.x') }}
-    # NumPy has a hard upper limit.
-    # NumPy 1.22.0, 1.22.1, 1.22.2 are all broken for ufuncs
-    # xref: https://github.com/numba/numba/issues/7756
-    - numpy >={{ numpy }},!=1.22.0,!=1.22.1,!=1.22.2,<1.26
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:52]: could you please explain why we remove numpy from run ?

Comment (cbouss) [reply recipe/meta.yaml:52]: It's still there, two lines below:
`- {{ pin_compatible('numpy', upper_bound='1.27') }}`
Upstream changed the minimum bound of numpy to `>=1.22,<1.27,!=1.22.0,!=1.22.1,!=1.22.2`, so I updated host and run accordingly.

Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:52]: does it mean we have for py 38, 39, 310 the numpy 1.22.3 in runtime ? Does not conflict what we have in global cbc.yaml ?

**Review by markan — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml:52]: The numpy pin set in the cbc is there to set a default minimum version, which defines our highest range supported (i.e. >=1.21,<2 for py38...310). At run time most users will get the highest compatible numpy version (1.26).
Here upstream, specifically set the minimum version to 1.22, so we have to follow as anyway their package is not compatible with 1.21.

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Pull request merged**